include/bredis/impl/connection.ipp

```diff
@@ -174,4 +149,19 @@ Connection<NextLayer>::read(DynamicBuffer &rx_buff,
     return boost::get<positive_parse_result_t>(parse_result);
 }
 
+template <typename NextLayer>
+template <typename DynamicBuffer>
+inline positive_parse_result_t
+Connection<NextLayer>::read(DynamicBuffer &rx_buff) {
+    namespace asio = boost::asio;
+    namespace sys = boost::system;
+
+    boost::system::error_code ec;
+    auto result = read(rx_buff, ec);
+    if (ec) {
+        throw ec;
```

Comment:
I believe you want
```
throw boost::system::system_error{ec}
```

See:
https://github.com/boostorg/asio/blob/develop/include/boost/asio/detail/impl/throw_error.ipp#L29